examples/autoignition-mpi.py

```diff
@@ -244,67 +298,144 @@ def main(ctx_factory=cl.create_some_context, casename="autoignition", use_leap=F
+            health_error = True
+            logger.info(f"{rank=}: Invalid pressure data found.")
+
+        if check_range_local(discr, "vol", dv.temperature, 1.4e3, 3.3e3):
+            health_error = True
+            logger.info(f"{rank=}: Invalid temperature data found.")
+
+        if dt < 0:
+            health_error = True
+            if rank == 0:
+                logger.info("Global DT is negative!")
+
+        return health_error
+
     def my_rhs(t, state):
         return (euler_operator(discr, cv=state, t=t,
                                boundaries=boundaries, eos=eos)
                 + eos.get_species_source_terms(state))
 
-    def my_checkpoint(step, t, dt, state):
-        if check_step(step, nrestart) and step != restart_step:
-            rst_filename = (
-                restart_path
-                + restart_file_pattern.format(casename=casename, step=step,
-                                              rank=rank)
-            )
-            rst_data = {
-                "local_mesh": local_mesh,
-                "state": current_state,
-                "t": t,
-                "step": step,
-                "global_nelements": global_nelements,
-                "num_parts": nproc
-            }
-            from mirgecom.restart import write_restart_file
-            write_restart_file(actx, rst_data, rst_filename, comm)
-
-        # awful - computes potentially expensive viz quantities
-        #         regardless of whether it is time to viz
-        reaction_rates = eos.get_production_rates(state)
-        viz_fields = [("reaction_rates", reaction_rates)]
-        return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              vizname=casename, step=step,
-                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
-                              constant_cfl=constant_cfl, comm=comm,
-                              viz_fields=viz_fields)
-
-    try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
-                get_timestep=get_timestep, state=current_state,
-                t=current_t, t_final=t_final)
-    except ExactSolutionMismatch as ex:
-        error_state = True
-        current_step = ex.step
-        current_t = ex.t
-        current_state = ex.state
-
-    if not check_step(current_step, nviz):  # If final step not an output step
-        if rank == 0:
-            logger.info("Checkpointing final state ...")
-        my_checkpoint(current_step, t=current_t,
-                      dt=(current_t - checkpoint_t),
-                      state=current_state)
-
-    if current_t - t_final < 0:
-        error_state = True
-
-    if error_state:
-        raise ValueError("Simulation did not complete successfully.")
+    def my_post_step(step, t, dt, state):
+        # Logmgr needs to know about EOS, dt, dim?
+        # imo this is a design/scope flaw
```

Comment:
I agree with this comment

Reply:
Fixed in #257 (well, except for `dt`; I haven't looked at what the logmgr does with it. @matthiasdiener ?)